lisa/notifiers/text_result.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Any, List, Type, Union
+
+from lisa import notifier, schema
+from lisa.runner import print_results
+from lisa.testsuite import TestResult, TestResultMessage
+from lisa.util import constants
+
+
+class TextResult(notifier.Notifier):
+    """
+    Creating log notifier to dump text formatted results for easier
+    view in editing mode. The original log is complete but too long to
+    check only the summary.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return constants.NOTIFIER_TEXT_RESULT
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return schema.Notifier
+
+    def _received_message(self, message: TestResultMessage) -> None:
+        if message.is_completed:
+            self.received_messages.append(Union(None, message))
+
+    def _subscribed_message_type(self) -> List[Type[notifier.MessageBase]]:
+        return [TestResultMessage]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self.result_file = open(
+            "{constants.RUN_LOCAL_PATH}/lisa-{constants.RUN_ID}-result.txt",
+            "x",
+        )
+        self.received_messages: List[Union[TestResult, TestResultMessage]] = []
+
+    def finalize(self) -> None:
+        print_results(self.received_messages, self.result_file.write)
```

Comment:
Because the file is write only in finalize method. So it can be opened in finalize now. use the `with` pattern to open it.